src/main/kotlin/com/angels/challenge/entity/StoreEntity.kt

```diff
@@ -0,0 +1,26 @@
+package com.angels.challenge.entity
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import java.io.Serializable
+import javax.persistence.*
+
+@Entity
+@Table(name="store")
+class StoreEntity() : Serializable {
+    @Id
+    @Column(name = "name")
+    var name: String = ""
```

Comment:
Can you imagine any disadvantage of having an id column having a string?

What approach are you using here? Natural keys?

Reply:
Here my thought process was that the name of the store was unique and hence could be used as primary key. Maybe it would have been more robust to include and auto generated ID instead as the primary key, so that if in the future our data would contain stores with the same name (and another property to distinguish between them) this could be more easy to deal on the dev part

Reply:
Yes. From my experience using surrogate IDs is the easiest ways with tools like Hibernate and other ORMs. Natural keys get usually mapped as indexes having unique constraints when applicable.